phc-rails/app/controllers/api/v1/accounts_controller.rb

```diff
@@ -1,31 +1,55 @@
 class Api::V1::AccountsController < ApplicationController
   respond_to :json
+  skip_before_action :verify_authenticity_token
 
   def search
-    if request.headers["AuthToken"].eql? ENV["AuthToken"]
-      first_name = request.headers["FirstName"]
-      last_name = request.headers["LastName"]
-      fhash = { first_name: first_name }
-      lhash = { last_name: last_name }
-      res = PersonAccount.fuzzy_search(fhash, false).fuzzy_search(lhash, false)
-      respond_with res
+    auth_token = request.headers["HTTP_AUTH_TOKEN"]
+    user_id = request.headers["HTTP_USER_ID"]
+    if auth_token && user_id
+      if user_authenticated?(user_id, auth_token)
+        first_name = request.headers["HTTP_FIRSTNAME"]
+        last_name = request.headers["HTTP_LASTNAME"]
+        result = PersonAccount.fuzzy_search({first_name: first_name}, false).fuzzy_search({last_name: last_name}, false)
```

Comment:
I would actually change what I did here from fuzzy searching in the controller to doing most of the logic in the model.